packages/frontend2/src/app/_components/chart/costs/scaling-costs-chart.tsx

```diff
@@ -0,0 +1,117 @@
+'use client'
+
+import { type Milestone } from '@l2beat/config'
+import React from 'react'
+import { useCostsTimeRangeContext } from '~/app/(new)/(other)/scaling/costs/_components/costs-time-range-context'
+import { useCostsUnitContext } from '~/app/(new)/(other)/scaling/costs/_components/costs-unit-context'
+import { Chart } from '~/app/_components/chart/core/chart'
+import { ChartProvider } from '~/app/_components/chart/core/chart-provider'
+import { RadioGroup, RadioGroupItem } from '~/app/_components/radio-group'
+import { Skeleton } from '~/app/_components/skeleton'
+import { useLocalStorage } from '~/hooks/use-local-storage'
+import { type CostsUnit } from '~/server/features/scaling/costs/types'
+import { api } from '~/trpc/react'
+import { useChartLoading } from '../core/chart-loading-context'
+import { CostsChartHover } from './costs-chart-hover'
+import { CostsChartTimeRangeControls } from './costs-chart-time-range-controls'
+import { useCostChartRenderParams } from './use-cost-chart-render-params'
+
+interface Props {
+  milestones: Milestone[]
+  tag?: string
+}
+
+export function ScalingCostsChart({ milestones, tag = 'costs' }: Props) {
+  const [scale, setScale] = useLocalStorage(`${tag}-scale`, 'lin')
```

Comment:
I know it's not related to this PR, but shouldn't we verify what's stored in the local storage in case we change the legal values?

Reply:
probably we should